test/benchmark_http_client_test.cc

```diff
@@ -26,6 +26,27 @@ using namespace testing;
 
 namespace Nighthawk {
 
+namespace {
+
+// Default function for request generator when the content doesn't matter.
+RequestGenerator getDefaultRequestGenerator() {
+  RequestGenerator request_generator = []() {
+    auto header = std::make_shared<Envoy::Http::TestRequestHeaderMapImpl>(
+        std::initializer_list<std::pair<std::string, std::string>>(
+            {{":scheme", "http"}, {":method", "GET"}, {":path", "/"}, {":host", "localhost"}}));
+    return std::make_unique<RequestImpl>(header);
+  };
+  return request_generator;
+}
+// Helper function to get headers in a set that should be verified during the test.
+absl::flat_hash_set<std::string>
+getTestRecordedProperties(const Envoy::Http::RequestHeaderMap& header) {
```

Comment:
I'm a bit confused by purpose and/or naming here. I'm wondering why does returns a `absl::flat_hash_set` which only has a single entry (path?), and create a `std::vector<absl::flat_hash_set>` later on? 
If the purpose is to record all paths requested throughout a test, would storing all the paths in a single `absl::flat_hash_set` instance be more convenient?

Reply:
That makes sense. I had originally wanted to do this with more properties, but in the end, it wasn't necessary.